client/widgets/CArtifactHolder.h

```diff
@@ -29,7 +29,7 @@ class CArtifactHolder
 	virtual void artifactAssembled(const ArtifactLocation & artLoc)=0;
 };
 
-class CArtPlace : public LRClickableAreaWTextComp
+class CArtPlace : public LRClickableAreaWTextComp, public SelectableSlot
```

Comment:
Do you mean `: virtual public CIntObject` ?
SelectableSlot and LRClickableAreaWTextComp use CIntObject internally. Wouldn't we get two instances of CIntObject in the case without `virtual` ?

Reply:
No, I mean use composition instead of inheriting from SelectableSlot:

```cpp
 class CArtPlace : public LRClickableAreaWTextComp
{
    std::shared_ptr<SelectableSlot> slot;
...
```
In this case you won't need virtual inheritance at all and SelectableSlot can inherit CIntObject without virtual specifier

(or swap LRClickableAreaWTextComp with SelectableSlot and inherit from SelectableSlot and have LRClickableAreaWTextComp as member, but this might be more complex)

Reply:
I don't think it would be correct to mix LRClickableAreaWTextComp and SelectableSlot. They have nothing in common except a common parent.

As for the composition: I plan to use SelectableSlot for the market as well. It will be needed a lot of additional wrappers for `selectSlot`, `isSelected` for each related class. Wouldn't this be code bloat?
Or perhaps
```
public:
  std::shared_ptr<SelectableSlot> slot;
```
is ok?